[Feature] 유저 퀴즈 세트 조회 기능 구현

Hut234 opened the pull request:
## 🔖 연관된 이슈
<!-- 연관된 이슈 번호를 작성해 주세요 -->
#60

## 📂 작업 내용
<!-- 작업한 내용을 작성해 주세요 -->
- 유저 퀴즈 세트 조회 기능을 위한 Entity, Controller, Service, Repository 생성
- 테스트 코드 작성
- user_quiz_set, user_quiz Flyway 파일 작성
-  시큐리티 관련 수정
    - 트랜잭션 추가
    - SpEL 수정   


## 📢 리뷰 참고사항
<!-- 리뷰시 참고사항을 작성해 주세요 -->

Hut234 committed:
feature: Add UserQuizSet, UserQuiz Entity

Hut234 committed:
feature: Rename duplicated name

Hut234 committed:
feature: Add controller, service, repository for get user quiz set

Hut234 committed:
test: Add get user quiz set test code

Hut234 committed:
chore: Add Flyway migration file for User_Quiz_Set, User_Quiz table

Hut234 committed:
feature: Add @Transactional(readOnly = true) for lazy loading

Hut234 committed:
fix: use #root instead of #authentication in SpEL

POKUDING commented on src/main/java/com/readup/server/user_quiz/domain/model/UserQuiz.java:
```diff
@@ -0,0 +1,57 @@
+	@Column(name = "id")
+	private Long id;
+
+	@Column(nullable = false)
```
이하 부분들도 column 명 명시해주시면 감사하겠습니다!

Hut234 replied:
반영하였습니다.

luz315 commented on src/main/java/com/readup/server/user_quiz/domain/model/UserQuiz.java:
```diff
@@ -0,0 +1,57 @@
+	private Long id;
+
+	@Column(nullable = false)
+	private Long quizId;
```
해당 필드가 고유하다면 unique = true 옵션이 필요할 것 같습니다 :)

Hut234 replied:
quizId와 createdBy를 묶어서 UK 적용하였습니다.

luz315 commented on src/main/java/com/readup/server/user_quiz/domain/model/UserQuizSet.java:
```diff
@@ -0,0 +1,72 @@
+	private Long id;
+
+	@Column(nullable = false)
+	private Long quizSetId;
```
해당 필드가 고유하다면 unique = true 옵션이 필요할 것 같습니다 :)

Hut234 replied:
quizSetId와 createdBy를 묶어서 UK 적용하였습니다.

luz315 commented on src/main/java/com/readup/server/user_quiz/domain/model/UserQuiz.java:
```diff
@@ -0,0 +1,57 @@
+
+@Getter
+@Entity
+@Builder
```
클래스 전체 빌더는 프로젝트 초기에는 효율성이 좋지만 안정성을 위해 전체가 아닌 제한된 필드에만 @Builder(access = AccessLevel.PRIVATE)로 묶어 정적메서드를 생성하는 방식도 공부해보심 좋을 것 같습니다 :)

Hut234 replied:
필요한 부분을 참고하여 반영하였습니다.

luz315 commented on src/main/java/com/readup/server/user_quiz/application/UserQuizSetService.java:
```diff
@@ -0,0 +1,50 @@
+
+	private UserQuizSet createNewUserQuizSet(Long quizSetId, List<Quiz> quizList) {
+		UserQuizSet newUserQuizSet = UserQuizSet.create(quizSetId);
+		quizList.forEach(quiz -> newUserQuizSet.addUserQuiz(quiz.getId()));
```
해당 부분의 코드 관점을 도메인으로 바꿔보면 좋을 것 같습니다
서비스가 직접 addUserQuiz()를 반복해서 호출하면 UserQuizSet 내부 상태와 동작 구조를 알게 됩니다
아래 방식도 고려해보시면 좋을 것 같습니다 :)

1. 아래와같이 도메인 내부에 상세코드를 명시합니다
`
public void addAllQuizzes(List<Quiz> quizList) {
	if (quizList == null || quizList.isEmpty()) return;
	quizList.forEach(quiz -> this.addUserQuiz(quiz.getId()));
}
`
2. 서비스는 도메인 측에 책임을 위임하고 단순 실행만 하게됩니다
`UserQuizSet quizSet = UserQuizSet.create(id);
quizSet.addAllQuizzes(quizList);`

Hut234 replied:
필요한 부분을 참고하여 반영하였습니다.

POKUDING reviewed (approved):
작업하시느라 고생하셨습니다!!
column 명만 명시해 주시면 좋을 것 같아요!

Hut234 committed:
feat: Add UK constraints

Hut234 committed:
feat: Add column names to entity classes

Hut234 committed:
refactor
- Remove redundant builder for constructor with fewer than 3 distinct-typed parameters
- Encapsulate UserQuizSet creation logic

Hut234 committed:
Merge branch 'development' into feature/#60-user-quiz-set-get
